airflow/models/abstractoperator.py

```diff
@@ -564,7 +566,7 @@ def _do_render_template_fields(
                     f"{attr_name!r} is configured as a template field "
                     f"but {parent.task_type} does not have this attribute."
                 )
-            if not value:
+            if isinstance(value, pd.DataFrame) or not value:
```

Comment:
Do we have a list of supported types we expect to use as `template_fields`? An alternative to skipping `pandas.DataFrame`, would be to check if the value type is within the supported types.

Reply:
@dstandish thank you very much for the feedback! My first attempt was to do `if value is not None`, but that broke a couple of tests. I'll try the other suggestions and let you know when it's ready for re-review.

Reply:
Yes. @dstandish is righ - we should not have pandas in Airflow core - Airflow does not like Pandas (even though most people do like Pandas). 

However maybe better is to simply try/catch ValueError in this place?  Contrary to many languages handling exception in Python is not slow, and it's pretty normal to handle different flow by handling exceptions

Reply:
Is there any place where we think actually raising an exception here would help? I think just continuing on ValueError would make it more resilient to any problematic classes that are ambiguous whether they are truthful or not.

Reply:
I like @dstandish 's solution here. We know that pandas/numpy/Series all have "empty" functions, and that anything I can think of that isn't "truthy" has a .empty function.

Reply:
Fine for me :)